openfe/setup/ligand_network_planning.py

```diff
@@ -228,6 +228,64 @@ def generate_minimal_spanning_network(
 
     return min_network
 
+def generate_minimal_redundant_network(
+    ligands: Iterable[SmallMoleculeComponent],
+    mappers: Union[AtomMapper, Iterable[AtomMapper]],
+    scorer: Callable[[LigandAtomMapping], float],
+    progress: Union[bool, Callable[[Iterable], Iterable]] = True,
+) -> LigandNetwork:
+    """
+    Plan a network with as few edges as possible with maximum total score, 
+    ensuring that every node is connected to two edges to introduce 
+    statistical redundancy.
+
+    Parameters
+    ----------
+    ligands : Iterable[SmallMoleculeComponent]
+      the ligands to include in the LigandNetwork
+    mappers : AtomMapper or Iterable[AtomMapper]
+      the AtomMapper(s) to use to propose mappings.  At least 1 required,
+      but many can be given, in which case all will be tried to find the
+      highest score edges
+    scorer : Scoring function
+      any callable which takes a LigandAtomMapping and returns a float
+    progress : Union[bool, Callable[Iterable], Iterable]
+      progress bar: if False, no progress bar will be shown. If True, use a
+      tqdm progress bar that only appears after 1.5 seconds. You can also
+      provide a custom progress bar wrapper as a callable.
+    """
+    if isinstance(mappers, AtomMapper):
+        mappers = [mappers]
+    mappers = [_hasten_lomap(m, ligands) if isinstance(m, LomapAtomMapper)
+               else m for m in mappers]
+
+    # First create a network with all the proposed mappings (scored)
+    network = generate_maximal_network(ligands, mappers, scorer, progress)
+
+    # Flip network scores so we can use minimal algorithm
+    g2 = nx.MultiGraph()
+    for e1, e2, d in network.graph.edges(data=True):
+        g2.add_edge(e1, e2, weight=-d['score'], object=d['object'])
+ 
+    # As in .generate_minimal_spanning_network(), use nx to get the minimal 
+    # network. But now also remove those edges from the fully-connected
+    # network, then get the minimal network again. Add mappings from all
+    # minimal networks together.
+    mappings = []
+    for _ in range(2): # can increase range here for more redundancy
```

Comment:
Maybe make the value of this range a parameter of the function, with default 2? That also allows a nice test, since you should get the existing MST network when the value is 1.

Reply:
great idea, added a test for this as well.